packages/codegen-ui/lib/utils/form-component-metadata.ts

```diff
@@ -27,12 +27,20 @@ export const getFormFieldStateName = (formName: string) => {
   return [formName.charAt(0).toLowerCase() + formName.slice(1), 'Fields'].join('');
 };
 
+/**
+ * @returns true if string contains special characters except for "." , "_" and 0-9
+ */
+export const isValidVariableName = (input: string): boolean => /^[a-zA-Z_$][a-zA-Z_$0-9]*$/g.test(input);
```

Comment:
sorry last minute ask, can we get a set of unit tests against these regexes to validate that they behave as expected? It's also easier this way to see if there may be any edge cases that may have been missed.

Reply:
Yup. sure thing